tests/backends/test_backends.py

```diff
@@ -388,3 +390,56 @@ def test_braket_parametric_cphase() -> None:
     equivalent_state(
         run(block, values=values, backend="braket"), run(block, values=values, backend="pyqtorch")
     )
+
+
+@pytest.mark.parametrize("backend_name", [BackendName.PYQTORCH, BackendName.HORQRUX])
+def test_dagger_returning_kernel(backend_name: BackendName) -> None:
+    def wf_is_normalized(wf: torch.Tensor) -> torch.Tensor:
+        return torch.isclose(sum(torch.flatten(torch.abs(wf) ** 2)), torch.tensor(1.00))
+
+    generatorx = 3.1 * X(0) + 1.2 * Y(0) + 1.1 * Y(1) + 1.9 * X(1) + 2.4 * Z(0) * Z(1)
+    fmx = HamEvo(generatorx, parameter=sympy.acos(Parameter("x")))
+    set_trainable(fmx, False)
+    fmy = HamEvo(generatorx, parameter=sympy.acos(Parameter("y")))
+    set_trainable(fmy, False)
+    ansatz = hea(2, 2)
+    set_trainable(ansatz, True)
+    circ = QuantumCircuit(2, fmx, ansatz.dagger(), ansatz, fmy.dagger())
+    backend = backend_factory(backend=backend_name, diff_mode=DiffMode.AD)
+    (pyqtorch_circ, _, embed, params) = backend.convert(circ)
+
+    initial_state = torch.rand((1, 2**2), dtype=torch.cdouble) + 1j * torch.rand(
+        (1, 2**2), dtype=torch.cdouble
+    )
+    initial_state = initial_state / torch.sqrt(4 * sum(abs(initial_state) ** 2))
+    inputs = {"x": torch.tensor([0.52]), "y": torch.tensor(0.52)}
+    if backend_name == BackendName.HORQRUX:
+        initial_state = tensor_to_jnp(initial_state)
+        inputs = {k: tensor_to_jnp(v) for k, v in inputs.items()}
+    run_params = embed(params, inputs)
+    wf = backend.run(pyqtorch_circ, run_params, state=initial_state)
+    if backend_name == BackendName.HORQRUX:
+        assert horq_equivalent_state(wf, initial_state)
+    else:
+        assert wf_is_normalized(wf)
+        assert equivalent_state(wf, initial_state)
+
+
+@pytest.mark.parametrize("n_qubits", [2, 3, 4])
+def test_compare_hevos(n_qubits: int) -> None:
+    register = Register.line(n_qubits)
+
+    gen = hamiltonian_factory(
+        register,
+        interaction=Interaction.ZZ,  # Parameterize for ZZ and XY
```

Comment:
Can you make this also a test parameter? For `Interaction.ZZ` and `Interaction.XY`. Should be fine but just to double-check with something non diagonal.

Reply:
added